tests/pipeline/test_adjusted_array.py

```diff
@@ -745,3 +745,42 @@ def test_update_labels(self):
         # values and adjustment values.
         check_arrays(adj_array.data, expected_adj_array.data)
         self.assertEqual(adj_array.adjustments, expected_adj_array.adjustments)
+
+    @parameterized.expand([
```

Comment:
This is a good start, but I think there are a few important cases we should probably cover:

1. Initial adjustments have a key not present in `adjustments_to_append`. It should still show up in the output.
2. Initial adjustments has more than one adjustment at a given index.
3. `adjustments_to_append` has more than one adjustment at a given index.